### DISCUSSION TITLE
MagicLink Sign in how to know if user already signed up

### DISCUSSION THREAD

**Discussion by socramm9** [supabase/supabase#12020] [Questions]
I am only allowing a password less sign up flow for my app. 
Users can be created using `signInWithOtp` if `shouldCreateUser` is set to true. 

But how do I trow an error if a user that already has an account tries to use the signUp flow that is bit different from my sign in flow.

And for `verifyOTP` the `OtpType` is needed. (signup or magiclink) but `signInWithOtp` dosen't let you know if the user was just created.

My current solution was to use `signUp` to handle the sign up flow. But the drawback is that I have to set a user password myself.

```
Future<void> signUpWithOtp({
    required String email,
    required String fullName,
  }) async {
    await api.auth.signUp(
      email: email,
      password: randomString(40),
      emailRedirectTo: kIsWeb ? null : 'app://login',
      data: {
        "full_name": fullName,
      },
    );
  }
```

**Comment by hqasmei** [comment]
Yes, I am having this exact same issue.
